### PR TITLE
archiver 3.3.0 (new formula)

### PR DESCRIPTION
**Pull request by vladimyr**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

**Review by dawidd6 — DISMISSED** [review]

File: Formula/archiver.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    # Don't set GOPATH because we want to build using go modules to
+    # ensure our dependencies are the ones specified in `go.mod`.
+    ENV["CGO_ENABLED"] = "0"
+    mkdir_p buildpath
+    ln_sf buildpath, buildpath/"archiver"
+
+    cd "archiver" do
+      system "go", "build", "-o", bin/"arc",
+             "cmd/arc/main.go"
+    end
```
Comment (dawidd6) [inline Formula/archiver.rb]:
```suggestion
    system "go", "build", "-trimpath", "-o", bin/"arc", "cmd/arc/main.go"
```

Comment (vladimyr) [reply Formula/archiver.rb]: @dawidd6 Thank you for your suggestion! It is pretty much evident that I'm not fluent in go and this was in fact just c/p from `termshark` formula 🙈 
I would accept it straight away but shouldn't we keep cgo disable switch (line 13) due to: https://github.com/mholt/archiver/blob/44285f7/build.bash#L6 ?

Comment (dawidd6) [reply Formula/archiver.rb]: I don't think it's completely necessary, but okay, let's keep it if upstream does too.

Comment (vladimyr) [reply Formula/archiver.rb]: I was thinking the same. Kept it just for the sake of compatibility.

Comment (Bo98) [reply Formula/archiver.rb]: It's probably for portability reasons. For example, if you compile against libc on Ubuntu and then want you use it on Alpine you might have issues, which `CGO_ENABLED=0` can avoid.

Comment (vladimyr) [reply Formula/archiver.rb]: Removed `CGO_ENABLED` var ✅

**Review by dawidd6 — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]

**Pull request closed without merging**